### PR TITLE
Added recording rules to metrics allowlist & refactored Grafana dashboards

### PR DESCRIPTION
**Pull request by dislbenn**
Related issue: https://github.com/stolostron/backlog/issues/19142
Signed-off-by: dislbenn <lavontae.bennett@gmail.com>

### CODE REVIEW

File: operators/multiclusterobservability/manifests/base/config/metrics_allowlist.yaml [hunk 1]
```diff
@@ -167,6 +167,24 @@ data:
...
+        expr: sum(container_memory_rss{endpoint=\"https-metrics\", container!=\"\"})
+      - record: container_memory_swap:sum
+        expr: sum(container_memory_swap{endpoint=\"https-metrics\", container!=\"\"})
+      - record: container_memory_working_set_bytes:sum
+        expr: sum(container_memory_working_set_bytes{endpoint=\"https-metrics\", container!=\"\"})
+      - record: kube_pod_container_resource_limits:sum
+        expr: sum(kube_pod_container_resource_limits{endpoint=\"https-main\", container!=\"\"})
+      - record: kube_pod_container_resource_requests:sum
+        expr: sum(kube_pod_container_resource_requests{endpoint=\"https-main\", container!=\"\"})
+      - record: kube_resourcequota:hard:scalar
```
Comment (marcolan018) [inline operators/multiclusterobservability/manifests/base/config/metrics_allowlist.yaml:186]: kube_resourcequota:hard:scalar
which high level dashboard requires this metrics? 
I can only find the metrics kube_resourcequota in Kubernetes / Compute Resources / Namespace (Pods) and Kubernetes / Compute Resources / Namespace (Workloads), both them are the dashboards for container/pod

File: operators/multiclusterobservability/manifests/base/config/metrics_allowlist.yaml [hunk 2]
```diff
@@ -167,6 +167,24 @@ data:
...
+      - record: container_cpu_cfs_throttled_periods_total:5m:sum
+        expr: sum(increase(container_cpu_cfs_throttled_periods_total{endpoint=\"https-metrics\", container!=\"POD\"}[5m]))
+      - record: container_memory_cache:sum
+        expr: sum(container_memory_cache{endpoint=\"https-metrics\", container!=\"\"})
+      - record: container_memory_rss:sum
+        expr: sum(container_memory_rss{endpoint=\"https-metrics\", container!=\"\"})
+      - record: container_memory_swap:sum
+        expr: sum(container_memory_swap{endpoint=\"https-metrics\", container!=\"\"})
+      - record: container_memory_working_set_bytes:sum
+        expr: sum(container_memory_working_set_bytes{endpoint=\"https-metrics\", container!=\"\"})
```
Comment (marcolan018) [inline operators/multiclusterobservability/manifests/base/config/metrics_allowlist.yaml:181]: for those memory related recording rules.
I only find container_memory_rss is used in some cluster level dashboards

File: operators/multiclusterobservability/manifests/base/config/metrics_allowlist.yaml (hunk 1 above)
Comment (dislbenn) [reply operators/multiclusterobservability/manifests/base/config/metrics_allowlist.yaml]: Correct, I removed `kube_resourcequota:hard:scalar`.

File: operators/multiclusterobservability/manifests/base/config/metrics_allowlist.yaml [hunk 3]
```diff
@@ -167,6 +167,24 @@ data:
         expr: sum(sum(sum(kube_pod_container_resource_requests{resource=\"memory\"}) by (pod,namespace,container) * on(pod,namespace) group_left(phase) max(kube_pod_status_phase{phase=~\"Running|Pending|Unknown\"} >0) by (pod,namespace,phase)) by (pod,namespace,phase))
       - record: sli:apiserver_request_duration_seconds:trend:1m
         expr: sum(increase(apiserver_request_duration_seconds_bucket{job=\"apiserver\",service=\"kubernetes\",le=\"1\",verb=~\"POST|PUT|DELETE|PATCH\"}[1m])) / sum(increase(apiserver_request_duration_seconds_count{job=\"apiserver\",service=\"kubernetes\",verb=~\"POST|PUT|DELETE|PATCH\"}[1m]))
+      - record: container_cpu_cfs_periods_total:5m:sum
+        expr: sum(increase(container_cpu_cfs_periods_total{endpoint=\"https-metrics\", container!=\"POD\"}[5m]))
+      - record: container_cpu_cfs_throttled_periods_total:5m:sum
```
[the remark this answers is not in the document]
Comment (dislbenn) [reply operators/multiclusterobservability/manifests/base/config/metrics_allowlist.yaml]: Correct, these recording rules: `container_cpu_cfs_periods_total:5m:sum ` and `container_cpu_cfs_throttled_periods_total:5m:sum` are now removed from the list.

File: operators/multiclusterobservability/manifests/base/config/metrics_allowlist.yaml (hunk 2 above)
Comment (dislbenn) [reply operators/multiclusterobservability/manifests/base/config/metrics_allowlist.yaml]: Yes, I have removed the following recording rules: `container_memory_cache:sum`, `container_memory_swap:sum`, and `container_memory_working_set_bytes:sum`.

**Comment by marcolan018** [comment]
@dislbenn 
also need to use those recording rule to replace the original metrics in the dashboards

File: operators/multiclusterobservability/manifests/base/config/metrics_allowlist.yaml [hunk 4]
```diff
@@ -177,11 +183,6 @@ data:
           - __name__="kube_pod_container_resource_limits",namespace="{{ $labels.namespace }}"
           - __name__="kube_pod_container_resource_requests",namespace="{{ $labels.namespace }}"
           - __name__="kube_resourcequota",namespace="{{ $labels.namespace }}"
-          - __name__="mixin_pod_workload",namespace="{{ $labels.namespace }}"
-          - __name__="namespace_workload_pod:kube_pod_owner:relabel",namespace="{{ $labels.namespace }}"
-          - __name__="node_namespace_pod_container:container_cpu_usage_seconds_total:sum_irate",namespace="{{ $labels.namespace }}"
-          - __name__="node_namespace_pod_container:container_cpu_usage_seconds_total:sum_rate",namespace="{{ $labels.namespace }}"
-
```
Comment (dislbenn) [inline operators/multiclusterobservability/manifests/base/config/metrics_allowlist.yaml:184]: Removed recording rules from the collection rules list. These will still be captured within the default metrics list.

File: operators/multiclusterobservability/manifests/base/config/metrics_allowlist.yaml [hunk 5]
```diff
@@ -193,9 +194,6 @@ data:
           - __name__="kube_pod_container_resource_limits",namespace="{{ $labels.namespace }}"
           - __name__="kube_pod_container_resource_requests",namespace="{{ $labels.namespace }}"
           - __name__="kube_resourcequota",namespace="{{ $labels.namespace }}"
-          - __name__="mixin_pod_workload",namespace="{{ $labels.namespace }}"
-          - __name__="namespace_workload_pod:kube_pod_owner:relabel",namespace="{{ $labels.namespace }}"
```
Comment (dislbenn) [inline operators/multiclusterobservability/manifests/base/config/metrics_allowlist.yaml:197]: Removed recording rules from the collection rules list. These will still be captured within the default metrics list.

File: operators/multiclusterobservability/manifests/base/config/metrics_allowlist.yaml [hunk 6]
```diff
@@ -208,11 +206,6 @@ data:
           - kube_pod_container_resource_limits 
           - kube_pod_container_resource_requests 
           - kube_resourcequota 
-          - mixin_pod_workload 
-          - namespace_workload_pod:kube_pod_owner:relabel 
-          - node_namespace_pod_container:container_cpu_usage_seconds_total:sum_irate 
-          - node_namespace_pod_container:container_cpu_usage_seconds_total:sum_rate 
-
```
Comment (dislbenn) [inline operators/multiclusterobservability/manifests/base/config/metrics_allowlist.yaml:215]: Removed recording rules from the collection rules list. These will still be captured within the default metrics list.

File: operators/multiclusterobservability/manifests/base/config/metrics_allowlist.yaml [hunk 7]
```diff
@@ -227,8 +220,6 @@ data:
           - kube_pod_container_resource_limits 
           - kube_pod_container_resource_requests 
           - kube_resourcequota 
-          - mixin_pod_workload 
-          - namespace_workload_pod:kube_pod_owner:relabel
```
Comment (dislbenn) [inline operators/multiclusterobservability/manifests/base/config/metrics_allowlist.yaml:231]: Removed recording rules from the collection rules `SNOHighMemoryUsage` list. These will still be captured within the default metrics list.

**Pull request merged**